### DISCUSSION TITLE
Is the automatic test for seasonality sufficiently robust?

### DISCUSSION THREAD

**Discussion by acartro** [pycaret/pycaret#2767] [Q&A] [answered]
Hi,

I am doing a time series project, and most of my TS with monthly granularity do not have seasonality, according to the pycaret test.

My question is if I really shouldn't put sp=12 to my time series and really let pycaret use as sp the primary seasonality, for all my series.

I have seen that in the grid search it applies a rule of sp=[sp, 2*sp]. If my series really have no seasonality, this would be the correct search grid ?

Another question, for the linear regression, that you have predefined to do a deseasonalisation, if you do sp=1, you are not really deseasonalising the series, are you?

Thanks

**Comment by ngupta23** [comment, marked as answer]
> My question is if I really shouldn't put sp=12 to my time series and really let pycaret use as sp the primary seasonality, for all my series.

That is essentially what `pycaret` does right now. For monthly data, it will test with for seasonality of 12. If seasonality is detected, it will use sp = 12 in the models. If it is not, it will not use seasonality in the models.

> I have seen that in the grid search it applies a rule of sp=[sp, 2*sp]. If my series really have no seasonality, this would be the correct search grid ?

I think when seasonality is not detected, there should be a branch that just uses sp = 1. In some models there is a conditional deseasonalizer which will only deseasonalize if the seasonality is detected. In such cases, it is ok to pass [sp, 2*sp] to such models as deseasonalizing only happens when seasonality is detected.

> Another question, for the linear regression, that you have predefined to do a deseasonalisation, if you do sp=1, you are not really deseasonalising the series, are you?

Correct, setting sp = 1 essentially means there is no seasonality.
